nano/node/confirmation_solicitor.cpp

```diff
@@ -1,62 +1,94 @@
 #include <nano/node/confirmation_solicitor.hpp>
-
 #include <nano/node/election.hpp>
 #include <nano/node/node.hpp>
 
+using namespace std::chrono_literals;
+
 nano::confirmation_solicitor::confirmation_solicitor (nano::node & node_a) :
-node (node_a)
+node (node_a),
+max_confirm_req_batches (node_a.network_params.network.is_test_network () ? 1 : 20),
+max_block_broadcasts (node_a.network_params.network.is_test_network () ? 4 : 30),
+min_time_between_requests (node_a.network_params.network.is_test_network () ? 10ms : 3s),
+min_time_between_floods (node_a.network_params.network.is_test_network () ? 20ms : 6s),
+min_request_count_flood (node_a.network_params.network.is_test_network () ? 0 : 4)
 {
 }
 
-void nano::confirmation_solicitor::prepare ()
+void nano::confirmation_solicitor::prepare (std::vector<nano::representative> const & representatives_a)
 {
 	assert (!prepared);
 	requests.clear ();
 	rebroadcasted = 0;
-	representatives = node.rep_crawler.representatives ();
```

Comment:
It looks like this class only needs `network` now (`network_params` could be made a member), so could remove the  `node` dependency on this class entirely. I guess not essential (only if bored) there's a generic ticket somewhere for me to do this :D

Reply:
Thanks, very good point and I should try addressing this more as I go (might try to for distributed_work when I head in there again).  `network_constants` are retrieved in the constructor itself and added a `network` reference as a member.